test/configuration_server_test.erl

```diff
@@ -0,0 +1,24 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%%
+%%% @end
+%%% Created : 16. May 2020 8:04 PM
+%%%-------------------------------------------------------------------
+-module(configuration_server_test).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-include("configuration_server.hrl").
+
+configuration_server_test_() ->
+    [
+        ?_assertEqual(?CONFIG_PATHS, configuration_server:default_directories()),
+        ?_assertEqual(
+            {error, {no_file, "No config file found"}},
+            configuration_server:config_file_name()
+        ),
+        ?_assertEqual(
+            {stop, {no_file, "No config file found"}},
+            configuration_server:init([])
+        )
```

Comment:
Considering this is a unit test I'm not too sure about these. Especially given that they're dependent on the system they're running on. Probably move them to a `SUITE`; either way I think we ought to mock the `?CONFIG_PATHS` if that's possible in Erlang.

Reply:
Maybe we could try to mock the functions in  `?CONFIG_PATH` and the necessary functions from `filelib` and `filenames` with the mocking library **[meck](http://eproxus.github.com/meck)**. Whats your opinion on that?

Reply:
I have added a simple mocking version for the `config_file_name()` function.